sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -701,7 +699,8 @@ ProgramManager::collectDeviceImageDepsForImportedSymbols(
     HandledSymbols.insert(ISProp->Name);
   }
   ur::DeviceBinaryType Format = MainImg.getFormat();
-  if (!WorkList.empty() && !checkLinkingSupport(Dev, MainImg))
+  if (!WorkList.empty() &&
+      !checkLinkingSupport(*getSyclObjImpl(Dev).get(), MainImg))
```

Comment:
Why aren't we changing line 691 instead?

Reply:
Because I didn't want to refactor the `doesDevSupportDeviceRequirements` function that internally calls the `checkDevSupportDeviceRequirements` that uses `tempalte <typename Param> device::get_info<Param>`

Reply:
Ok, I'll look into it separately.